redaxo/src/core/lib/form/form.php

```diff
@@ -28,7 +28,10 @@ class rex_form extends rex_form_base
     protected $whereCondition;
     /** @var string */
     protected $mode;
-    /** @var int */
+    /**
+     * @var int
+     * @psalm-var positive-int
```

Comment:
Vielleicht noch eine allgemeine Anmerkung: Ich bin noch nicht soo überzeugt von der Nutzung von positive-int, jedenfalls solange wir es als zusätzliche `psalm-`-Zeile schreiben.
Am konkreten Beispiel: Ich halte es für nicht wirklich wahrscheinlich, dass wir irgendwo versehentlich direkt einen negativen Int verwenden, und psalm den somit erkennt. Wenn schon eher, dass er aus irgendeiner Quelle kommt, die für psalm so aber auch nicht ersichtlich ist.

Und hier im PR sieht man ja, was das für Kreise zieht, wo man dann überall zusätzliche Docs notieren muss.

Musst jetzt hier nicht alles wieder rückgängig machen. Ist schon ok so. Wollte es nur anmerken, dass ich nicht so Fan davon bin, wenn wir jetzt überall, wo es positve Ints sind (was ja zum Beispiel IDs immer sind, somit wird es noch sehr viele Stellen geben), das als zusätzliche Docs notieren.

Reply:
Im beispiel hier beginnt man bei 1 zu zählen was wir nicht oft haben. Daher find ich es sinnvoll. 

Ich wollte demnächst nen FOR trick schreiben wie man statische analyse für seine addons nutzen kann.. damit wird das coverage hoffentlich mehrwert auch in anderen repositories bringen

Reply:
Normale IDs fangen doch eigentlich alle mit 1 an.

Reply:
Also ich meine sowas wie `$articleId` etc.

Reply:
Ja das schon. Aber 90% der php nativen datenstrukturen beginnen bei 0 und dann kann man da einfach nen off-by-one hinbekommen